src/systems/handle_action_points.rs

```diff
@@ -0,0 +1,29 @@
+use crate::*;
+
+system!(
+    HandleActionPointsSystem,
+    |entities: Entities<'a>,
+    stats: WriteStorage<'a, Comp<StatSet<Stats>>>,
+    simple_movements: ReadStorage<'a, SimpleMovement>,
+    hero1_simple_movements: ReadStorage<'a, Hero1SimpleMovement>,
+    targets: WriteStorage<'a, AiDestination>| {
+        for (e, _) in (&*entities, &simple_movements).join() {
```

Comment:
I think it would be better to add the stats storage inside of the join instead of having many unwraps.

Reply:
That makes sense. I have done so.